subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/VersionResolver.groovy

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.performance.fixture
+
+import com.google.common.base.Splitter
+import groovy.transform.CompileStatic
+import groovy.transform.TypeCheckingMode
+import org.gradle.integtests.fixtures.executer.GradleDistribution
+import org.gradle.integtests.fixtures.versions.ReleasedVersionDistributions
+import org.gradle.util.GradleVersion
+
+import java.util.regex.Pattern
+
+class VersionResolver {
+
+    private static final Pattern COMMA_OR_SEMICOLON = Pattern.compile('[;,]')
+
+    static List<String> resolveBaselineVersions(String overrideBaselinesProperty = System.getProperty('org.gradle.performance.baselines'), List<String> targetVersions = []) {
```

Comment:
To me it seems like this shouldn't be a default parameter. Maybe have another method for getting this from the environment? Or when moving `toBaselineVersions` here this may also be easier to deal with. In other words, no class outside of this class should obtain this system property.

Reply:
pls check if that's what you meant

Reply:
It is better now, though `BuildScanPerformanceTest` still references the system property. I think the system property should only be referenced in `BaselineVersionResolver`, and maybe exposed by it if necessary, but no other class should call `System.getProperty('org.gradle.performance.baselines')`.

Reply:
done